scene/resources/surface_tool.cpp

```diff
@@ -1152,9 +1152,19 @@ void SurfaceTool::mikktSetTSpaceDefault(const SMikkTSpaceContext *pContext, cons
 }
 
 void SurfaceTool::generate_tangents() {
-	ERR_FAIL_COND(!(format & Mesh::ARRAY_FORMAT_TEX_UV));
 	ERR_FAIL_COND(!(format & Mesh::ARRAY_FORMAT_NORMAL));
-
+	if (!(format & Mesh::ARRAY_FORMAT_TEX_UV)) {
```

Comment:
Is it a requirement to have tangents, some of the collada code generates tangents, but I have a desire to not generate them if wrong.

Reply:
Tangents are always required, if the mesh doesn't provide them then the renderer will generate them. Ideally tangent generation would happen at import time though